specs/governance/alligator.md

```diff
@@ -0,0 +1,207 @@
+# Alligator
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Interface](#interface)
+  - [Core Functions](#core-functions)
+    - [`subdelegate`](#subdelegate)
+    - [`subdelegateFromToken`](#subdelegatefromtoken)
+    - [`subdelegateBatched`](#subdelegatebatched)
+    - [`afterTokenTransfer`](#aftertokentransfer)
+  - [Getters](#getters)
+    - [`getSubdelegations`](#getsubdelegations)
+    - [`getCheckpoints`](#getcheckpoints)
+    - [`getVotingPower`](#getvotingpower)
+- [Storage](#storage)
+- [Types](#types)
+  - [`SubdelegationRule`](#subdelegationrule)
+  - [`AllowanceType`](#allowancetype)
+  - [`Checkpoint`](#checkpoint)
+- [Backwards Compatibility](#backwards-compatibility)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Constant | Value                                        |
+|----------|----------------------------------------------|
+| Address  | `0x4200000000000000000000000000000000000043` |
+
+The `Alligator` contract implements subdelegations for the [`GovernanceToken`](gov-token.md). Subdelegations enable
+advanced delegation use cases, such as partial, time-constrained & block-based delegations, and relative & fixed allowances.
+
+The `Alligator` contract migrates the delegation state from the [`GovernanceToken`](gov-token.md) to itself
+through a hook-based approach. Specifically, the [`GovernanceToken`](gov-token.md) calls the `Alligator` contract's
+`afterTokenTransfer` function after a token transfer. This enables the `Alligator` contract to consume the hook and update
+its delegation and checkpoint mappings accordingly. If either address involved in the transfer (`_from_` or `_to`) has not
+been migrated to the `Alligator` contract, the contract copies the address' checkpoint data from the
+[`GovernanceToken`](gov-token.md) to its own state.
+
+## Interface
+
+### Core Functions
+
+#### `subdelegate`
+
+Allows subdelegation of token voting power to another address (delegatee) with specified subdelegation rules. This function
+is inteded to be called by users that require advanced delegation of the [`GovernanceToken`](gov-token.md).
+
+```solidity
+subdelegate(address _delegatee, SubdelegationRule _rule)
+```
+
+#### `subdelegateFromToken`
```

Comment:
Now that `subdelegate` has had its function signature changed, do we need to name the other function `subdelegateFromToken` or can it be an overloaded `subdelegate`?

Reply:
imo it's more clear if it's named `subdelegateFromToken` because of the limitation that it can only be called from the `GovernanceToken`